src/main/java/org/jboss/weld/examples/command/CommandExtension.java

```diff
@@ -50,120 +43,28 @@ void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager beanMana
         event.addContext(commandContext);
 
         // Register the command context bean
-        event.addBean(new Bean<CommandContext>() {
-
-            @Override
-            public CommandContext create(CreationalContext<CommandContext> creationalContext) {
-                return new InjectableCommandContext(commandContext, beanManager);
-            }
-
-            @Override
-            public void destroy(CommandContext instance, CreationalContext<CommandContext> creationalContext) {
-            }
-
-            @Override
-            public Set<Type> getTypes() {
-                return ImmutableSet.of(CommandContext.class);
-            }
-
-            @Override
-            public Set<Annotation> getQualifiers() {
-                return ImmutableSet.of(DefaultLiteral.INSTANCE);
-            }
-
-            @Override
-            public Class<? extends Annotation> getScope() {
-                return Dependent.class;
-            }
-
-            @Override
-            public String getName() {
-                return null;
-            }
-
-            @Override
-            public Set<Class<? extends Annotation>> getStereotypes() {
-                return Collections.emptySet();
-            }
-
-            @Override
-            public boolean isAlternative() {
-                return false;
-            }
-
-            @Override
-            public Class<?> getBeanClass() {
-                return CommandExtension.class;
-            }
-
-            @Override
-            public Set<InjectionPoint> getInjectionPoints() {
-                return Collections.emptySet();
-            }
-
-            @Override
-            public boolean isNullable() {
-                return false;
-            }
-        });
+        event.addBean()
```

Comment:
This could be even less verbose since setting things empty set is pointless (leavning that out will have same result) - but then again, as learning example, we might want to preserve it. WDYT?